### ISSUE TITLE
Handle PK changes

### ISSUE THREAD

**Issue by awm33** [datamill-co/target-postgres#15]
The only thing we may be able to do is warn and update it, or fail hard. PK changes could be an actual source change, or more likely the using had it wrong in the Singer stream.

https://github.com/datamill-co/target-postgres/blob/master/target_postgres/postgres.py#L65

**Comment by AlexanderMann** [comment]
I'm going to start in on this by failing hard. This will force us to _detect_ that the keys have changed, then we can go about implementing the rest of things.

**Comment by awm33** [comment]
@AlexanderMann Keep in mind, we don't actually declare it as a primary key. Couple reasons for that: 1) Other data warehouse systems, such as Redshift, do not support actual primary keys. 2) What we're calling primary key or "key properties" is really just a set of unique columns used for loading and dedupping.

It's just stored in the comment metadata along with the table version.

**Comment by AlexanderMann** [comment]
So I didn't realize that we don't actually _create_ primary keys right now, instead we make comments on the tables. I can easily revamp a bunch of the code I've already written, but this does bring up an interesting question:

Do we want to create primary keys on tables?

**Comment by AlexanderMann** [comment]
Yup, I didn't see that till after your comment. I can revamp a bunch of the code I have written.

**Comment by AlexanderMann** [comment]
Given Stitch actually creates PKs in PostgreSQL for this, is that potentially something we'll want to do?

**Comment by AlexanderMann** [comment]
@awm33 yeah, saw that in their docs. See [here for an example]:

> Table is created with Primary Key constraint (table_name_pkey). Primary Key column has NOT NULL constraint.

Looks like that's just a difference between the PSQL and Redshift implementations. Not something we _have_ to do. Just a point to weigh.
